### PR TITLE
Astral

### PR DESCRIPTION
**Pull request by isgiddy**
- [x] Closes #44
 - [x] Tests added
 - [x] Passes `pre-commit run --all-files`
 - [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

-----------
Started over.  
Creating a new pull request to handle issue #44.  
I'm submitting the PR maybe early but I think reviews will help it get off the ground quicker..especially thanks to @jbusecke  for all the support so far!  

This function now implements skyfield.
I have added a catch for polar night/polar day.

Currently passes the test I created, but appreciate feedback to make this more robust.

### CODE REVIEW

**Comment by callumrollo** [comment]
Checking the CI logs, failures in 3.6 and 3.8, looks like a warning is raised when the plot test is run. Could be something to check? This test was only introduced in #118 so it may have unintended consequences we haven't seen before. If the warning is caused by astral, we may need to change this test to be better targeted at plotting.

**Comment by jbusecke** [comment]
I am editing the test function over in #123, so that we can see what warning is actually raised. Or is this visible in the pytest output right now (I could not find anything helpful).

File: tests/test_optics.py [hunk 1]
```diff
@@ -1,55 +1,32 @@
...
-    """
-    import numpy as np
-
-    import glidertools as gt
-
-    time = [
-        np.datetime64("2000-12-21"),
-        np.datetime64("2000-06-21"),
-    ]
-    lat = (
```
Comment (jbusecke) [inline tests/test_optics.py:45]: Is 75 deg lat (above), sufficient to test that this behavior is now working?

File: tests/test_optics.py [hunk 2]
```diff
@@ -1,55 +1,32 @@
 import pytest
 
 
-def test_sunrise_sunset():
+def test_sunrise_sunset_skyfield():
```
Comment (jbusecke) [inline tests/test_optics.py:4]: I would leave the name as is. You are not testing anything specific to the implementation?, just the results of the glidertools function, so the name might induce confusion.

File: tests/test_optics.py [hunk 3]
```diff
@@ -1,55 +1,32 @@
...
-    import glidertools as gt
+    from glidertools.optics import sunset_sunrise
 
     time = [
         np.datetime64("2000-01-01"),
         np.datetime64("2000-01-02"),
         np.datetime64("2000-01-03"),
     ]
-    lat = -35, 35, 45
+    lat = -35, 75, 45
```
Comment (jbusecke) [inline tests/test_optics.py:21]: See below: I was wondering why not to use 80 (as in the previously failing test)?

File: tests/test_optics.py [hunk 4]
```diff
@@ -1,55 +1,32 @@
...
     lon = 0, 0, 0
-    sunrise, sunset = gt.optics.sunset_sunrise(time, lat, lon)
+    sunrise, sunset = sunset_sunrise(time, lat, lon)
 
-    # Two entries, there should be two outputs
+    # Three entries, there should be three outputs
     assert len(sunrise) == len(lat)
 
     # sunrise will be earlier for the SH in January
     assert sunrise[0] < sunrise[1]
```
Comment (jbusecke) [inline tests/test_optics.py:29]: Can we add more precise expected values? Not sure this is necessary?

File: tests/test_optics.py [hunk 5]
```diff
@@ -1,55 +1,32 @@
...
-    )
-    lon = (
-        0,
-        0,
-    )
-
-    with pytest.raises(ValueError):
-        sunrise, sunset = gt.optics.sunset_sunrise(time, lat, lon)
+    # high latitude in NH winter should output polar night
+    assert pd.to_datetime(sunrise[1]).hour == 11
```
Comment (jbusecke) [inline tests/test_optics.py:32]: Oh I didnt see this. That looks good. Maybe assert for the other two values too?

**Review by jbusecke — COMMENTED** [review]
This looks great @isgiddy! Thank you very much. I had a few comments (all minor). 
From my end this looks ready to merge, but it might be good to get another review from someone @GliderToolsCommunity/developers to look over this.

File: tests/test_optics.py (hunk 1 above)
Comment (isgiddy) [reply tests/test_optics.py:45]: yes, the fail started from ~65 , but I can add a higher latitude value too? Not necessary?

**Comment by jbusecke** [comment]
Can you try to `git pull`. I did rebase the branch onto the newest `master` (I should have made this more clear, sorry for the trouble).

**Comment by callumrollo** [comment]
Sorry to keep pestering, but could we get this PR merged and have a fresh release ready for the demo on Monday?

File: tests/test_optics.py (hunk 1 above)
Comment (jbusecke) [reply tests/test_optics.py:45]: No thats fine in my view. Thanks for clarifying.

**Review by jbusecke — APPROVED** [review]
Lets get this moving! @isgiddy do you mind if I add the entry to `whats-new.rst`? That way I can get this out of the door today and @callumrollo has a happy Monday!

**Comment by jbusecke** [comment]
Ill go ahead and merge. Big thanks to @isgiddy! 👏

**Pull request merged**